plutus-tx/src/PlutusTx/Builtins.hs

```diff
@@ -202,6 +215,276 @@ verifySchnorrSecp256k1Signature
 verifySchnorrSecp256k1Signature vk msg sig =
   fromBuiltin (BI.verifySchnorrSecp256k1Signature vk msg sig)
 
+-- | Converts an 'Integer' into its 'BuiltinByteString' representation.
+--
+-- = Notes
+--
+-- Throughout, let @maxInteger@ be the maximum function for 'Integer's,
+-- and @absInteger@ be the absolute value function for 'Integer's.
+-- We define @zeroes :: 'Integer' -> 'BuiltinByteString'@ be the
+-- function which, given input @i@, produces a 'BuiltinByteString' @bs@ such
+-- that:
+--
+-- * @'lengthByteString' bs@ @=@ @'maxInteger' 0 i@; and
+-- * For all @j :: 'Integer'@ such that @'greaterThanEqualsInteger' j 0@,
+-- @'indexByteString' bs j = 0@;
+--
+-- We define @ones :: 'Integer' -> 'BuiltinByteString'@ identically to @zeroes@,
+-- except that @'indexByteString' bs j = 255@ instead.
+--
+-- == Laws
+--
+-- 'integerToByteString' must roundtrip via 'byteStringToInteger'. Specifically,
+-- for all @i :: 'Integer'@, we must have:
+--
+-- @'byteStringToInteger '.' 'integerToByteString' '$' i@ @=@ @i@
+--
+-- Furthermore, the length of any result of 'integerToByteString' must be
+-- strictly positive:
+--
+-- @'greaterThanInteger' ('lengthByteString' '.' 'integerToByteString' i) 0@ @=@
```

Comment:
FWIW, I think this section would be clearer if you just used the mathematical notation. I appreciate that you're trying to use the actual functions, which is nice in a way but I think overall less clear.